### PR TITLE
continuous-integration2: Python linting fixes

### PR DESCRIPTION
**Pull request by nathanchance**
See all the individual commits for the warnings and reasoning.

Depends on https://github.com/ClangBuiltLinux/actions-workflows/pull/4.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -30,9 +30,15 @@ jobs:
     - uses: actions/checkout@v3
     - name: check-patches.sh
       run: bash scripts/check-patches.sh
+  python:
+     strategy:
+       fail-fast: false
+       matrix:
+         version: ['3.11', '3.10', '3.9', '3.8']
```
Comment (nickdesaulniers) [inline .github/workflows/lint.yml:37]: I'm ok to drop support for python3 < 3.10.

**Review by nickdesaulniers — APPROVED** [review]
LGTM; minor thoughts but you need not fix them to land. :truck: :anatomical_heart:

File: generate_tuxsuite.py [hunk 2]
```diff
@@ -110,5 +110,5 @@ def emit_tuxsuite_yml(config, tree, llvm_version):
     # check command line flags.
     generated_config = get_config_from_generator()
     args = parse_args(generated_config["trees"])
-    for llvm_version in get_llvm_versions(generated_config, args.tree):
-        emit_tuxsuite_yml(generated_config, args.tree, llvm_version)
+    for llvm_ver in get_llvm_versions(generated_config, args.tree):
```
[the remark this answers is not in the document]
Comment (nathanchance) [reply generate_tuxsuite.py:113]: Whoops! I updated the message with the right warnings :)

File: .github/workflows/lint.yml (hunk 1 above)
Comment (nathanchance) [reply .github/workflows/lint.yml:37]: I don't really think it hurts anything to keep the older ones if we are not writing code that necessitates a newer version but I do not feel strongly about it so I can drop it if you insist, we are pretty much the only two people who work in this repo anyways.

**Pull request merged**